File: backend/internal/domain/ingredient.go
package domain

import "time"

type Ingredient struct {
	ID            uint
	Name          string
	Unit          string  // "kg", "L", "un", "g", "ml"
	StockQuantity float64 // quantidade atual em estoque
	MinStock      float64 // alerta de estoque mínimo (opcional)
	Active        bool    // soft delete flag
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

File: frontend/src/lib/types/ingredient.ts
export interface Ingredient {
  ID: number;
  Name: string;
  Unit: string;
  StockQuantity: number;
  MinStock: number;
  Active: boolean;
  CreatedAt?: string;
  UpdatedAt?: string;
}

export interface IngredientCreatePayload {
  name: string;
  unit: string;
  stock_quantity: number;
  min_stock: number;
}
